provider/aws/aws.go

```diff
@@ -505,17 +529,51 @@ func (p *AWSProvider) submitChanges(ctx context.Context, changes []*route53.Chan
 				params := &route53.ChangeResourceRecordSetsInput{
 					HostedZoneId: aws.String(z),
 					ChangeBatch: &route53.ChangeBatch{
-						Changes: b,
+						Changes: ownedRecordRemoved(b),
 					},
 				}
 
+				successfulChanges := 0
+
 				if _, err := p.client.ChangeResourceRecordSetsWithContext(ctx, params); err != nil {
-					log.Errorf("Failure in zone %s [Id: %s]", aws.StringValue(zones[z].Name), z)
-					log.Error(err) //TODO(ideahitme): consider changing the interface in cases when this error might be a concern for other components
-					failedUpdate = true
+					log.Errorf("Failure in zone %s [Id: %s] when submitting change batch", aws.StringValue(zones[z].Name), z)
+					log.Error(err)
+
+					changesByOwnership := groupChangesByNameAndOwnership(b)
+
+					if len(changesByOwnership) > 1 {
+						log.Error("Trying to submit change sets one-by-one instead")
+
+						for _, changes := range changesByOwnership {
+							for _, c := range changes {
+								log.Infof("Desired change: %s %s %s [Id: %s]", *c.Action, *c.ResourceRecordSet.Name, *c.ResourceRecordSet.Type, z)
+							}
+							params.ChangeBatch = &route53.ChangeBatch{
+								Changes: ownedRecordRemoved(changes),
+							}
+							if _, err := p.client.ChangeResourceRecordSetsWithContext(ctx, params); err != nil {
+								failedUpdate = true
+								log.Error("Failed submitting change, it will be retried in a separate change batch in the next iteration")
+								if _, ok := p.failedChangesQueue[z]; !ok {
+									p.failedChangesQueue[z] = changes
+								} else {
+									p.failedChangesQueue[z] = append(p.failedChangesQueue[z], changes...)
+								}
+							} else {
+								log.Info("Change(s) successful")
```

Comment:
This log line should be removed entirely.

Reply:
Fixed